### DISCUSSION TITLE
Parent and child at the same time

### DISCUSSION THREAD

**Discussion by lucix-mich** [netbox-community/netbox#6859] [Q&A] [answered]
Hello community,

I know device bays are not meant to model individual components in a main device. Still, we are using them that way (inventory items are too limited for us). Here, the point is not to enter into this discussion - separate discussion coming along for that ;-).

My point here is: would it be difficult to allow a device to be both parent and child? Other then being able to set these flags, would that have negative consquences for the device model? If not, I would formulate it as a feature request (and maybe look into it).

Thanks and cheers

Michel

**Comment by candlerb** [comment, marked as answer]
I am pretty sure that a request to turn parent/child into a recursive hierarchy would be rejected.

Child devices are meant to be fully-fledged, independent, manageable devices, with their own IP address and their own independent control planes.  They are *not* meant for things like line cards, and if you use them in that way, lots of things break (one example: LAGs between ports across line cards).  They are also not meant for things like SFPs or other modules.

This feature is only intended for where the parent is a chassis or dumb housing, and the children are independent devices like blade servers or modular patch panels.

**Reply by lucix-mich** [reply]
Hello Brian,

I know about the intention of device bays, which seems to currently be the only use of parent/child relations. I haven't researched or experienced all the limitations, so thanks for mentioning one.

But as can be seen from the recurring discussions about the same topic, namely providing a model for any type of pluggable hardware, there really seems to be a need.

Inventory items don't cut it - we need something that exists as a separate object, has a life of its own. Albeit not necessarily as a managed device.

For what it's worth, cable management (blocks rack units) and patch panels are also modeled as full-fledged devices, although in general they are not managed (haven't seen a managed cable guide at least ;-).

Thanks and cheers

Michel

**Reply by candlerb** [reply]
> Inventory items don't cut it - we need something that exists as a separate object, has a life of its own

I agree, but then that's the feature request you want: to be able to assign an inventory item to a rack or location (make the device assignment optional), and to be able to transfer an inventory item from one device to another.

Interestingly, the data model has for a long time allowed for recursive parent/child hierarchy in inventory items.  It originally wasn't exposed in the GUI (#1692) but as of v2.10 it should be.

Extending inventory items with interfaces and front/rearports would be useful too: #824
